--- sql/mysqld.h	2013-09-10 16:38:26.000000000 +0900
+++ sql/mysqld.h	2013-09-29 15:18:44.000000000 +0900
@@ -189,7 +189,7 @@
extern const double log_10[309];
extern ulonglong keybuff_size;
extern ulonglong thd_startup_options;
-extern ulong thread_id;
+extern MYSQL_PLUGIN_IMPORT ulong thread_id;
extern ulong binlog_cache_use, binlog_cache_disk_use;
extern ulong binlog_stmt_cache_use, binlog_stmt_cache_disk_use;
extern ulong aborted_threads;
--- sql/sql_cache.h	2013-09-10 16:38:26.000000000 +0900
+++ sql/sql_cache.h	2013-09-29 15:18:44.000000000 +0900
@@ -548,5 +548,5 @@
};
#define QUERY_CACHE_FLAGS_SIZE sizeof(Query_cache_query_flags)

-extern Query_cache query_cache;
+extern MYSQL_PLUGIN_IMPORT Query_cache query_cache;
#endif
--- sql/global_threads.h	2013-09-10 16:38:26.000000000 +0900
+++ sql/global_threads.h	2013-09-29 15:18:44.000000000 +0900
@@ -25,7 +25,7 @@
 
 class THD;
 
-extern mysql_mutex_t LOCK_thread_count;
+extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_thread_count;
extern mysql_mutex_t LOCK_thread_remove;
extern mysql_cond_t COND_thread_count;


--- include/mysql/service_mysql_alloc.h	2013-09-10 16:38:26.000000000 +0900
+++ include/mysql/service_mysql_alloc.h	2013-09-29 15:18:44.000000000 +0900

extern struct mysql_malloc_service_st *mysql_malloc_service;

-#ifdef MYSQL_DYNAMIC_PLUGIN
+#if (MYSQL_DYNAMIC_PLUGIN && !NO_USE_MALLOC_SERVICE_INTERFACE) 

#define my_malloc mysql_malloc_service->mysql_malloc
#define my_realloc mysql_malloc_service->mysql_realloc
