--- sql/mysqld.h
+++ sql/mysqld.h
@@ -148,7 +148,7 @@
 extern const double log_10[309];
 extern ulonglong keybuff_size;
 extern ulonglong thd_startup_options;
-extern ulong thread_id;
+extern MYSQL_PLUGIN_IMPORT ulong thread_id;
 extern ulong binlog_cache_use, binlog_cache_disk_use;
 extern ulong binlog_stmt_cache_use, binlog_stmt_cache_disk_use;
 extern ulong aborted_threads,aborted_connects;
@@ -208,7 +208,7 @@
 extern const char *load_default_groups[];
 extern struct my_option my_long_options[];
 int handle_early_options();
-extern int mysqld_server_started, mysqld_server_initialized;
+extern MYSQL_PLUGIN_IMPORT int mysqld_server_started, mysqld_server_initialized;
 extern "C" MYSQL_PLUGIN_IMPORT int orig_argc;
 extern "C" MYSQL_PLUGIN_IMPORT char **orig_argv;
 extern pthread_attr_t connection_attrib;
@@ -230,7 +230,7 @@
   THR_MALLOC is a key which will be used to set/get MEM_ROOT** for a thread,
   using my_pthread_setspecific_ptr()/my_thread_getspecific_ptr().
 */
-extern pthread_key(MEM_ROOT**,THR_MALLOC);
+extern MYSQL_PLUGIN_IMPORT pthread_key(MEM_ROOT**,THR_MALLOC);
 
 #ifdef HAVE_PSI_INTERFACE
 #ifdef HAVE_MMAP
@@ -501,8 +501,8 @@
 #ifdef HAVE_OPENSSL
 extern mysql_mutex_t LOCK_des_key_file;
 #endif
-extern mysql_mutex_t LOCK_server_started;
-extern mysql_cond_t COND_server_started;
+extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_server_started;
+extern MYSQL_PLUGIN_IMPORT mysql_cond_t COND_server_started;
 extern mysql_rwlock_t LOCK_grant, LOCK_sys_init_connect, LOCK_sys_init_slave;
 extern mysql_rwlock_t LOCK_system_variables_hash;
 extern mysql_cond_t COND_thread_count;
@@ -558,10 +558,11 @@ extern mysql_mutex_t
        LOCK_item_func_sleep, LOCK_status, LOCK_show_status,
        LOCK_error_log, LOCK_delayed_insert, LOCK_short_uuid_generator,
        LOCK_delayed_status, LOCK_delayed_create, LOCK_crypt, LOCK_timezone,
-       LOCK_slave_list, LOCK_active_mi, LOCK_manager,
+       LOCK_slave_list/*, LOCK_active_mi*/, LOCK_manager,
        LOCK_global_system_variables, LOCK_user_conn,
        LOCK_prepared_stmt_count, LOCK_error_messages, LOCK_connection_count,
        LOCK_slave_init;
+extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_active_mi;
 extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_thread_count;
 #ifdef HAVE_OPENSSL
 extern char* des_key_file;
--- sql/slave.h
+++ sql/slave.h
@@ -250,7 +250,7 @@ bool net_request_file(NET* net, const char* fname);
 extern bool volatile abort_loop;
 extern Master_info *active_mi; /* active_mi for multi-master */
 extern Master_info *default_master_info; /* To replace active_mi */
-extern Master_info_index *master_info_index;
+extern MYSQL_PLUGIN_IMPORT Master_info_index *master_info_index;
 extern LEX_STRING default_master_connection_name;
 extern my_bool replicate_same_server_id;
 
--- sql/sql_cache.h
+++ sql/sql_cache.h
@@ -596,5 +596,5 @@
 #define query_cache_is_cacheable_query(L) 0
 #endif /*HAVE_QUERY_CACHE*/
 
-extern Query_cache query_cache;
+extern MYSQL_PLUGIN_IMPORT Query_cache query_cache;
 #endif
