--- sql/mysqld.h	2013-09-11 21:10:41.000000000 +0900
+++ sql/mysqld.h	2014-12-05 19:25:31.000000000 +0900
@@ -186,7 +186,7 @@
 extern const double log_10[309];
 extern ulonglong keybuff_size;
 extern ulonglong thd_startup_options;
-extern ulong thread_id;
+extern MYSQL_PLUGIN_IMPORT ulong thread_id;
 extern ulong binlog_cache_use, binlog_cache_disk_use;
 extern ulong binlog_stmt_cache_use, binlog_stmt_cache_disk_use;
 extern ulong aborted_threads;
@@ -273,7 +273,7 @@
 extern struct my_option my_long_early_options[];
 int handle_early_options();
 void adjust_related_options(ulong *requested_open_files);
-extern int mysqld_server_started;
+extern MYSQL_PLUGIN_IMPORT int mysqld_server_started;
 extern "C" MYSQL_PLUGIN_IMPORT int orig_argc;
 extern "C" MYSQL_PLUGIN_IMPORT char **orig_argv;
 extern pthread_attr_t connection_attrib;
@@ -744,9 +744,9 @@
 #ifdef HAVE_OPENSSL
 extern mysql_mutex_t LOCK_des_key_file;
 #endif
-extern mysql_mutex_t LOCK_server_started;
+extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_server_started;
 extern mysql_mutex_t LOCK_thread_created;
-extern mysql_cond_t COND_server_started;
+extern MYSQL_PLUGIN_IMPORT mysql_cond_t COND_server_started;
 extern mysql_rwlock_t LOCK_grant, LOCK_sys_init_connect, LOCK_sys_init_slave;
 extern mysql_rwlock_t LOCK_system_variables_hash;
 extern mysql_cond_t COND_manager;
--- sql/sql_cache.h	2013-09-11 21:10:41.000000000 +0900
+++ sql/sql_cache.h	2014-12-05 19:04:20.000000000 +0900
@@ -545,5 +545,5 @@
 };
 #define QUERY_CACHE_FLAGS_SIZE sizeof(Query_cache_query_flags)
 
-extern Query_cache query_cache;
+extern MYSQL_PLUGIN_IMPORT Query_cache query_cache;
 #endif
--- sql/global_threads.h	2013-09-11 21:10:41.000000000 +0900
+++ sql/global_threads.h	2014-12-05 19:03:55.000000000 +0900
@@ -22,7 +22,7 @@
 
 class THD;
 
-extern mysql_mutex_t LOCK_thread_count;
+extern MYSQL_PLUGIN_IMPORT mysql_mutex_t LOCK_thread_count;
 extern mysql_mutex_t LOCK_thread_remove;
 extern mysql_cond_t COND_thread_count;
 
--- include/mysql/service_mysql_alloc.h	2013-12-03 12:09:53.000000000 +0900
+++ include/mysql/service_mysql_alloc.h	2014-12-05 19:27:11.000000000 +0900
@@ -49,7 +49,7 @@
 
 extern struct mysql_malloc_service_st *mysql_malloc_service;
 
-#ifdef MYSQL_DYNAMIC_PLUGIN
+#if (defined(MYSQL_DYNAMIC_PLUGIN) && !defined(NO_USE_MALLOC_SERVICE_INTERFACE))
 
 #define my_malloc mysql_malloc_service->mysql_malloc
 #define my_realloc mysql_malloc_service->mysql_realloc
